src/main/java/schemacrawler/crawl/StreamlinerSchemaCrawler.java

```diff
@@ -0,0 +1,259 @@
+            List<Map<String, Object>> result = Collections.synchronizedList(new ArrayList<>());
+            try {
+                List<Future> futures = new ArrayList<>();
+                for (int threadIndex = 0; threadIndex < numThreads; threadIndex++) {
+                    int finalThreadIndex = threadIndex;
+                    futures.add(executorService.submit(() -> {
+                        String queryString = query.replace("{{SCHEMA_NAME}}", schemaName)
+                                .replace("{{THREAD_INDEX}}", String.valueOf(finalThreadIndex));
+                        StopWatch timer = StopWatch.createStarted();
+                        String queryForLogging = query.replace('\n', ' ');
+                        try (Connection connection = connectionSupplier.get();
+                             Statement statement = connection.createStatement()) {
+                            try {
+                                // I don't believe this is required to perform parallel queries across different connections
+                                // but someone has suggested it's required, the query runs without incident, and has no
+                                // known performance impact, so we run it but continue on if it fails.
+                                statement.execute("begin " +
+                                        "execute immediate 'alter session set parallel_degree_policy=auto'; " +
+                                        "end;");
+                            } catch (SQLException e) {
+                                log.info("Unable to set parallel_degree_policy=auto, we believe this should not impact you", e);
+                            }
+                            log.info("Executing: " + queryForLogging);
+                            ResultSet rs = statement.executeQuery(queryString);
+                            ResultSetMetaData rsmd = rs.getMetaData();
+                            while (rs.next()) {
+                                int count = rsmd.getColumnCount();
+                                Map<String, Object> row = new LinkedHashMap<>();
+                                for (int i = 1; i < count + 1; i++) {
+                                    row.put(rsmd.getColumnLabel(i), rs.getObject(i));
+                                }
+                                log.trace(row);
+                                result.add(row);
+                            }
+                            timer.stop();
+                            log.info(String.format("Thread %s took %s to execute %s", finalThreadIndex, timer.formatTime(),
+                                    queryForLogging));
+                            return result;
+                        } catch (SQLException e) {
+                            throw new IllegalStateException(e);
+                        }
+                    }));
+                }
+                for (Future f : futures) {
```

Comment:
Is it not necessary to wait for all futures to complete? (I'm not very familiar with Java futures)

Reply:
I guess `get` must block

Reply:
Yep get blocks forever